Fix Bug 3431: Clean up usage of ActivityStreamPrefs to use Prefs from the store

SeanPrashad opened the pull request:
Fixes #3431: Update redundant `.get()` calls to ActivityStreamPrefs to use `this.store.getState().Prefs.values`. Also updating the respective unit tests to accomodate this new behaviour.

**Note**: Currently a WIP as of Jan 4, 2018

Mardak commented on system-addon/test/unit/lib/ManualMigration.test.js:
```diff
@@ -38,17 +42,23 @@ describe("ManualMigration", () => {
 
-    const {ManualMigration} = injector({"lib/ActivityStreamPrefs.jsm": {Prefs: fakePrefs}});
+    // const {ManualMigration} = injector({"lib/ActivityStreamPrefs.jsm": {Prefs: fakePrefs}});
+    const {ManualMigration} = injector({"lib/ActivityStreamPrefs.jsm": {}});
```
The `injector` was only needed to pass in a fake prefs, so because we're no longer using that, we should be able to just `import` ManualMigration as a regular module at the top of the test file.

Mardak commented on system-addon/lib/ManualMigration.jsm:
```diff
@@ -18,7 +17,22 @@ XPCOMUtils.defineLazyModuleGetter(this, "ProfileAge", "resource://gre/modules/Pr
+  }
+
+  set setMigrationRemainingDays(newDate) {
+    this.store.getState().Prefs.values.migrationRemainingDays = newDate;
```
We don't want to directly modify the store without notifying others. See `expireMigration` in this file.

Mardak commented on system-addon/lib/ManualMigration.jsm:
```diff
@@ -18,7 +17,22 @@ XPCOMUtils.defineLazyModuleGetter(this, "ProfileAge", "resource://gre/modules/Pr
+  }
+
+  get getMigrationRemainingDays() {
+    return this.store.getState().Prefs.values.migrationRemainingDays;
```
Thanks for cleaning this up. I think instead of having individual getter/setters for each pref, it would be simple enough to have a helper that takes in the pref name. Although at that point, it's nothing specific to manual migration, so maybe it should be available for all Feeds?

@k88hudson any thoughts on having `this.store.getPref(name)` and similar `setPref` helper that dispatches?

k88hudson replied:
I can see that being helpful, yeah; I guess it is a bit verbose to do `this.store.getState().Prefs.values` every time.

I think extending `store` to have `store.getPref` and `store.setPrefs` as helpers would be ok, but if it's out of the scope of this PR a single getter/setter in `ManualMigration.jsm` would be fine by me too.

SeanPrashad commented:
@Mardak I've updated the ManualMigrations `import` statement as requested and modified both setters to update the store as appropriate.

I'm not attempting to switch over the old `fakePrefs` statements in `ManualMigration.test.js` but I'm unsure what the old `fakePrefs.get.returns(today);` call would look like in the updated version - any suggestions?

k88hudson commented:
@SeanPrashad This is looking great so far! For the tests, instead of using the `fakePrefs` helper,  you'll want to set whatever value you need on your `instance.store.state` object. For example, instead of `fakePrefs.get.returns(today)`, you would set `instance.store.state.migrationLastShownDate = today`. Does that make sense?

SeanPrashad commented:
Thank you @k88hudson! That makes much more sense - I will try to take a crack at it sometime soon!

Mardak commented:
Thanks for putting together this initial PR! Superseded by #4071.

k88hudson commented:
Thanks for your work on this! Your patch landed in this commit https://github.com/mozilla/activity-stream/commit/b634b4b0a438c0ad1f51b9d20cc0fc25b6d8c26d with some additional test fixes here: (https://github.com/mozilla/activity-stream/commit/2c8e2422be4a1db5802bc20207d34a22a4aeb42f)
